### PR TITLE
Improving bg common logging

### PR DESCRIPTION
**Pull request by CarlosGamero**
## Checklist

- [x] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [x] I've updated the documentation, or no changes were necessary
- [x] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts [hunk 1]
```diff
@@ -89,19 +90,33 @@ export class BackgroundJobProcessorMonitor<
...
       },
       `Finished job ${job.name}`,
     )
     this.transactionObservabilityManager.stop(resolveJobId(job))
   }
+
+  private buildLogParams(job: JobType, error?: unknown) {
+    return {
+      origin: this.processorName,
+      jobId: resolveJobId(job),
```
Comment (kibertoad) [inline packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:114]: isn't jobId coming from the childLogger already?

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts [hunk 2]
```diff
@@ -89,19 +90,33 @@ export class BackgroundJobProcessorMonitor<
...
         isSuccess: job.progress === 100,
       },
       `Finished job ${job.name}`,
     )
     this.transactionObservabilityManager.stop(resolveJobId(job))
   }
+
+  private buildLogParams(job: JobType, error?: unknown) {
+    return {
+      origin: this.processorName,
```
Comment (kibertoad) [inline packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:113]: wonder if more of these fields should be moved to the childLogger building step

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts [hunk 3]
```diff
@@ -89,19 +90,33 @@ export class BackgroundJobProcessorMonitor<
   public jobStart(job: JobType, requestContext: RequestContext): void {
     const transactionName = `bg_job:${this.ownerName}:${this.queueId}`
     this.transactionObservabilityManager.start(transactionName, resolveJobId(job))
-    requestContext.logger.info({ origin: this.processorName }, `Started job ${job.name}`)
+    requestContext.logger.info(this.buildLogParams(job), `Started job ${job.name}`)
+  }
+
+  public jobTryError(job: JobType, error: unknown, requestContext: RequestContext): void {
```
Comment (kibertoad) [inline packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:96]: name is slightly confusing, what does `jobTryError` mean?

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 1 above)
Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:114]: you are completely right! I will revisit what is on child logger and remove from here, sorry 🙏

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 2 above)
Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:113]: Not really, origin is where is the log generated, if the request context is passed to a service/use case, origin should not be there from my point of view, but please feel free to disagree

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 3 above)
Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:96]: it is that the job throws an error on the process try, but it does not have to be a definitive error (the job can be retried depending on the config).

I agree that it is confusing, but I am super bad at naming, let me give it a thought but please if you have something in mind I am more than open to ideas 🙏

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 2 above)
Comment (kibertoad) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:113]: I don't mean the requestContext that we pass any deeper, just the logger that we instantiate for the job. I mean this part:

```ts
    const requestContext: RequestContext = {
      reqId: jobId,
      logger: new BackgroundJobProcessorLogger(
        this.logger.child({
          // add more stuff here, it won't propagate anywhere else
          'x-request-id': job.data.metadata.correlationId,
          jobId,
          jobName: job.name,
        }),
        job,
      ),
    }
```

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 3 above)
Comment (kibertoad) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:96]: Do you mean BullMQ "job attempt" or something else?

Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:96]: Yes, the job attempt, maybe `jobAttemtError` is meaningful enough?

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 2 above)
Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:113]: Yeah, but we should always use `requestContext.logger`, right?
If, from the `processor.process` method, we call to a use-case or service where we need to log something we should be passing `requestContext` and in that case origin is not correct

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 3 above)
Comment (kibertoad) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:96]: sounds good!

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 2 above)
Comment (kibertoad) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:113]: ah, fair point.
but isn't that already a problem?

Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts:113]: for other properties, it is ...  well I guess it's fine to put the job id or job name to see which was the "starter" of the process but we are using `origin`  to put the class from where the log is written

File: packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts (hunk 1 above)
Comment (CarlosGamero) [reply packages/app/background-jobs-common/src/background-job-processor/monitoring/BackgroundJobProcessorMonitor.ts]: Done, thanks 🙏

**Review by kibertoad — APPROVED** [review]

**Pull request merged**